File: Command/Command/ViewController.m
//
//  ViewController.m
//  Command
//
//  Created by ZhangYu on 2017/9/1.
//  Copyright © 2017年 DeveloperZhang. All rights reserved.
//

#import "ViewController.h"
#import "Invoker.h"
#import "ConcreteCommand.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    Command *command = [[ConcreteCommand alloc] initWithReceiver:[Receiver new]];
    Invoker *invoker = [[Invoker alloc] initWithCommand:command];
    [invoker executeCommand];
}

@end

File: Visitor/Visitor/ViewController.swift
//
//  ViewController.swift
//  Visitor
//
//  Created by ZhangYu on 2017/9/11.
//  Copyright © 2017年 DeveloperZhang. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        let o: ObjectStructure = ObjectStructure();
        o.attatch(element: ConcreteElementA());
        o.attatch(element: ConcreteElementB());
        let v1: ConcreteVisitor1 = ConcreteVisitor1();
        let v2: ConcreteVisitor2 = ConcreteVisitor2();
        o.accept(visitor: v1);
        o.accept(visitor: v2);
        
    }
}

